### PR TITLE
add margin loss function from scone

### PR DESCRIPTION
**Pull request by amztc34283**
Added loss function: EnergyMarginLoss in src/pytorch_ood/loss/scone.py
Added functions on utils.py: evaluate_classification_loss_training, evaluate_energy_logistic_loss
Added example on examples/loss/supervised/scone.py

### CODE REVIEW

**Comment by kkirchheim** [comment]
Thank you for the implementation. Could you maybe do the following adjustments?

- add some basic test, e.g. a forward and a backwards pass through the loss 
- add some documentation to the parameters of the loss 
- move the `evaluate_classification_loss_training` from the general utils to scone.py, as it is only used there?

**Comment by amztc34283** [comment]
Let me know if there is anything else you want me to fix!
